src/main/scala/io/citrine/lolo/bags/BaggedTrainingResult.scala

```diff
@@ -8,94 +8,78 @@ import scala.collection.parallel.immutable.ParSeq
-    if (predictedVsActual.nonEmpty) {
-      Some(loss)
-    } else {
-      None
+  override lazy val predictedVsActual: Option[Seq[(Vector[Any], Double, Double)]] = Some(
+    trainingData.zip(NibT).flatMap {
+      case (TrainingRow(f, l, _), nb) =>
+        val oob = if (disableBootstrap) {
+          ensembleModels.zip(nb)
+        } else {
+          ensembleModels.zip(nb).filter(_._2 == 0)
+        }
+
+        if (oob.isEmpty || l.isNaN) {
+          Seq.empty
+        } else {
+          val predicted = oob.map(_._1.transform(Seq(f)).expected.head).sum / oob.size
+          Seq((f, predicted, l))
+        }
     }
+  )
+
+  override lazy val loss: Option[Double] = predictedVsActual.collect {
+    case pva if pva.nonEmpty => RegressionMetrics.RMSE(pva)
   }
 }
 
 case class ClassificationBaggerTrainingResult[T](
     ensembleModels: ParSeq[Model[T]],
     Nib: Vector[Vector[Int]],
     trainingData: Seq[TrainingRow[T]],
-    featureImportance: Option[Vector[Double]],
+    override val featureImportance: Option[Vector[Double]],
     disableBootstrap: Boolean = false
 ) extends BaggedTrainingResult[T] {
 
   lazy val NibT: Vector[Vector[Int]] = Nib.transpose
-  lazy val model: BaggedClassificationModel[T] = BaggedClassificationModel(ensembleModels)
-  lazy val predictedVsActual: Seq[(Vector[Any], T, T)] = trainingData.zip(NibT).flatMap {
-    case (TrainingRow(f, l, _), nb) =>
-      val oob = if (disableBootstrap) {
-        ensembleModels.zip(nb)
-      } else {
-        ensembleModels.zip(nb).filter(_._2 == 0)
-      }
-
-      if (oob.isEmpty || l == null) {
-        Seq()
-      } else {
-        val predicted = oob.map(_._1.transform(Seq(f)).getExpected().head).groupBy(identity).maxBy(_._2.size)._1
-        Seq((f, predicted, l))
-      }
-  }
-
-  lazy val loss: Double = ClassificationMetrics.loss(predictedVsActual)
-
-  override def getFeatureImportance(): Option[Vector[Double]] = featureImportance
 
-  override def getModel(): BaggedClassificationModel[T] = model
-
-  override def getPredictedVsActual(): Option[Seq[(Vector[Any], T, T)]] = Some(predictedVsActual)
-
-  override def getLoss(): Option[Double] = {
-    if (predictedVsActual.nonEmpty) {
-      Some(loss)
-    } else {
-      None
+  override lazy val model: BaggedClassificationModel[T] = BaggedClassificationModel(ensembleModels)
+
+  override lazy val predictedVsActual: Option[Seq[(Vector[Any], T, T)]] = Some(
+    trainingData.zip(NibT).flatMap {
+      case (TrainingRow(f, l, _), nb) =>
+        val oob = if (disableBootstrap) {
+          ensembleModels.zip(nb)
+        } else {
+          ensembleModels.zip(nb).filter(_._2 == 0)
+        }
+
+        if (oob.isEmpty || l == null) {
+          Seq()
+        } else {
+          val predicted = oob.map(_._1.transform(Seq(f)).expected.head).groupBy(identity).maxBy(_._2.size)._1
+          Seq((f, predicted, l))
+        }
     }
+  )
+
+  override lazy val loss: Option[Double] = predictedVsActual.collect {
```

Comment:
This recreates the logic on `getLoss()` which only returns a value if `predictedVsActual` is non-empty